odxtools/odxlink.py

```diff
@@ -172,18 +173,31 @@ class OdxLinkDatabase:
     This can resolve ODXLINK references.
     """
 
-    def __init__(self) -> None:
+    def __init__(self, *, use_weakrefs: bool = False) -> None:
         self._db: dict[OdxDocFragment, dict[str, Any]] = {}
+        self.use_weakrefs = use_weakrefs
 
     @overload
-    def resolve(self, ref: OdxLinkRef, expected_type: None = None) -> Any:
+    def resolve(self,
+                ref: OdxLinkRef,
+                expected_type: None = None,
+                *,
+                use_weakrefs: bool | None = None) -> Any:
```

Comment:
Already in init, why in resolve as well?

Reply:
because I frequently use the `OdxLink` database to resolve references in my application code and when I wrote this code I expected to get normal references, i.e., it is sometimes necessary to change the default value of the `OdxLinkDatabase` object...

Reply:
a new `clone(self, use_weakrefs)` function would be better in that case